Jetpack WordPress.com Toolbar Accessibility improvements

ryelle opened the pull request:
**This PR is done against `add/masterbar-module`, in an effort to add accessibility before the toolbar is merged**

Improved the keyboard accessibility and screen reader experience when using the WordPress.com toolbar in Jetpack sites.

- Add screen reader text for the Notifications icon
- Added focus styling to all links
- Add default group explicitly so that the HTML menu order matches display (for more intuitive keyboard navigation)
- Add accessible-focus script from calypso

This PR also fixes a bug in Safari where the "My Sites" dropdown was hidden by a overzealous core style on `.my-sites`.

There are fixes that needed to happen in the WordPress.com code as well, so make sure to have D4639-code applied to your sandbox when testing.

**To test**

1. Check out this branch on your Jetpack site
2. Apply D4639-code to your sandbox
3. Make sure the WordPress.com Toolbar is enabled under Jetpack > Settings > General
4. Use your keyboard to navigate through the toolbar
5. Use a screen reader ([VoiceOver](https://href.li/?http://webaim.org/articles/voiceover/), [NVDA](https://href.li/?http://webaim.org/blog/testing-with-the-nvda-screen-reader/), [JAWS](https://href.li/?http://webaim.org/articles/jaws/)) to navigate through the toolbar

Getting into the notifications with VoiceOver was a little tricky, but I think it's an improvement from previously.

See #6413

ryelle commented:
This could also use a design review for the focus styles, I just quickly added a visible outline to all elements:

![screen shot 2017-03-07 at 5 16 08 pm](https://cloud.githubusercontent.com/assets/541093/23680589/c6571e24-0359-11e7-9171-ca9ae78bebb4.png)

![screen shot 2017-03-07 at 5 07 24 pm](https://cloud.githubusercontent.com/assets/541093/23680576/bbb6cfd2-0359-11e7-80f6-78f67ee3464e.png)

jeherve commented on modules/masterbar/masterbar.php:
```diff
@@ -146,6 +147,7 @@ public function add_notifications( $wp_admin_bar ) {
 		$wp_admin_bar->add_node( array(
 			'id'     => 'notes',
 			'title'  => '<span id="wpnt-notes-unread-count" class="wpnt-loading wpn-read"></span>
+						 <span class="screen-reader-text">Notifications</span>
```
Could you make that text translatable?

ryelle replied:
Done, thanks for catching that :)

ryelle commented:
@eliorivero & @jeherve Thanks for the feedback — I updated that string to be translated, and added the prefix to the script. I also updated D4639 with that padding suggestion :)

eliorivero commented:
LGTM 🐑

dereksmart commented:
Closing in favor of #6713 because I failed at rebasing this.
